### DISCUSSION TITLE
Confusion around FlowChannel1Phase position argument

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#21835] [Q&A Modules: General] [answered]
Hi MOOSE team, I am currently using the thermal hydraulics module for a project, and I am a bit confused about the position argument. In step 5 on the single phase flow example, when a second flow channel is added, its position argument on line 214 for the water flow channel is altered slightly to include the thickness of the wall, compared to the helium flow cahnnel. I have included a little drawing to show my understanding of the setup for this set of flow channels and heat exchanger. I don't understand why the position argument requires us to take into account the thickness of the wall, when it has the same "origin" as the rest of the components. My intuition for the geometry may be wrong which would explain why I don't get it, but I was just wondering if there was something else I'm missing. Any help is much appreciated!

Small edit, sorry, realised it was just rude to reference lines on one of the example input files and send anyone answering on a hunt for the correct file. I have attached the relevant file, but the parts I am referring to are this line for the helium flow channel

```
position = '1 0 2'
```

and this line for the water flow channel.

```
position = '${fparse 1 + hx_wall_thickness} 0 2'
```

![thermalHyd](https://user-images.githubusercontent.com/54721825/184354657-64b3566e-da70-443e-b726-63dab36a1ac1.png)
[example_5.txt](https://github.com/idaholab/moose/files/9316603/example_5.txt)
l

**Comment by GiudGiud** [comment]
@joshuahansel is the secondary flow supposed to wrap around the pipe (what it seems looking at A) or be on the other side of the wall (what it seems looking at the position argument) ?

Link to that tutorial:
https://mooseframework.inl.gov/modules/thermal_hydraulics/tutorials/single_phase_flow/step05.html#step-5-secondary-side

**Comment by joshuahansel** [comment, marked as answer]
Thanks for bringing to my attention @GiudGiud. @TheBEllis Sorry for the confusion in the example. The example does have an annular secondary side channel, and you're right, the axis line segment is what you're usually specifying with `position`, `orientation`, and `length`, so in reality, you shouldn't offset as in the example. The example offsets the secondary side for visualization purposes only: if you were to open up the output file in Paraview without the offset, the primary and secondary sides would be coincident, so the person who wrote the example put the secondary side flow channel just to the right of that intermediate heat structure so that they can be viewed simultaneously instead of having to toggle the primary and secondary side channels on and off. The important thing to realize for this is that few or no cases, this `position` doesn't actually matter. The only thing that really matters is that you have the channels aligned with other meshes correctly so that nearest node-type couplings work correctly.

**Reply by TheBEllis** [reply]
Thank you both very much for getting back to me and offering some clarification it was very very helpful! To rack your brains just a bit more, in both the step of the tutorial that @GiudGiud linked, as well as my own input files, I can't actually open the output files in paraview, as it doesn't like the fact that sidesets and element blocks have the same name? Have either of you had any experience of this?

**Reply by GiudGiud** [reply]
This is an incompatibility with your version of paraview, 5.10.something. 
You may either downgrade to 5.9.latest or use the legacy exodus reader in your version
